crates/external/rpc/src/traits.rs

```diff
@@ -51,16 +68,60 @@ pub trait ExecutingMessageValidator {
-        tokio::time::timeout(timeout, fut)
+        match tokio::time::timeout(timeout, fut)
             .await
             .map_err(ExecutingMessageValidatorError::ValidationTimeout)?
+        {
+            Ok(_) => Ok(()),
+            Err(err) => {
+                match err {
+                    ClientError::Call(err) => {
+                        // Check if it's invalid message call, message example:
+                        // `failed to check message: failed to check log: unknown chain: 14417`
+                        if err.message().contains(UNKNOWN_CHAIN_MSG) {
+                            if let Ok(chain_id) = err
+                                .message()
+                                .split(' ')
+                                .last()
+                                .expect("message contains chain id")
+                                .parse::<u64>()
+                            {
+                                return Err(ExecutingMessageValidatorError::UnknownChain(chain_id));
+                            }
+                        // Check if it's `does not meet the minimum safety` error, message example:
+                        // `message {0x4200000000000000000000000000000000000023 4 1 1728507701 901}
+                        // (safety level: unsafe) does not meet the minimum safety cross-unsafe"`
+                        } else if err.message().contains(MINIMUM_SAFETY_MSG) {
+                            let message_safety =
+                                if err.message().contains("safety level: finalized") {
+                                    SafetyLevel::Finalized
+                                } else if err.message().contains("safety level: safe") {
+                                    SafetyLevel::Safe
+                                } else if err.message().contains("safety level: local-safe") {
+                                    SafetyLevel::LocalSafe
+                                } else if err.message().contains("safety level: cross-unsafe") {
+                                    SafetyLevel::CrossUnsafe
+                                } else if err.message().contains("safety level: unsafe") {
+                                    SafetyLevel::Unsafe
+                                } else if err.message().contains("safety level: invalid") {
+                                    SafetyLevel::Invalid
+                                } else {
+                                    // Unexpected level name, return generic error
+                                    return Err(ExecutingMessageValidatorError::Other(err));
+                                };
+                            return Err(ExecutingMessageValidatorError::MinimumSafety {
+                                expected: safety,
+                                got: message_safety,
+                            });
+                        }
+                        Err(ExecutingMessageValidatorError::Other(err))
```

Comment:
let's extract this code to `impl From<ErrorObjectOwned> for ExecutingMessageValidatorError`, and then add a test for the conversion